Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -169,17 +179,23 @@ updateDriverDailyZscore ride rideDate driverZscore chargeableDistance merchantId
             case driverZscore of
               Nothing -> dailyLeaderBoardConfig.zScoreBase + getMeters (fromMaybe 0 chargeableDistance)
               Just zscore -> do
-                let (prevTotalRides, prevTotalDistance) = getRidesAndDistancefromZscore zscore dailyLeaderBoardConfig.zScoreBase
-                let currTotalRides = prevTotalRides + 1
-                let currTotalDist = prevTotalDistance + fromMaybe 0 chargeableDistance
-                currTotalRides * dailyLeaderBoardConfig.zScoreBase + getMeters currTotalDist
-      Hedis.zAddExp (makeDailyDriverLeaderBoardKey merchantId rideDate) ride.driverId.getId (fromIntegral currZscore) lbExpiry.getSeconds
+                let (prevTotalRides, prevTotalDistance, prevTotalValidCustomers, prevTotalReferrals) = getParamsfromZscore zscore dailyLeaderBoardConfig.zScoreBase driverLeaderBoardType
+                case driverLeaderBoardType of
+                  LConfig.RIDE -> do
+                    let currTotalRides = prevTotalRides + 1
+                    let currTotalDist = prevTotalDistance + fromMaybe 0 chargeableDistance
+                    currTotalRides * dailyLeaderBoardConfig.zScoreBase + getMeters currTotalDist
+                  LConfig.REFERRAL -> do
+                    let currTotalValidCustomers = prevTotalValidCustomers + (if validCustomer then 1 else 0)
+                    let currTotalReferrals = prevTotalReferrals + 1
```

Comment:
`validCustomer` check?

Reply:
we require to do the referral ranking according to the number of customers referred who has taken valid ride. That's why added valid customer check
first level sorting - total activated customers
second level sorting - total referred customers